### DISCUSSION TITLE
i have no idea what i am doing 😬

### DISCUSSION THREAD

**Discussion by seatownbttm** [CodyBerenson/PGMA-Modernized#226] [General]
hiya. 

i’m brand new to plex and was trying to find ways to add gay porn and came across this. 

i’ve been trying to figure out how to make this work but i know nothing about coding. i’m not sure i even know what i’m trying to do?

if i’m in over my head - let me know and i’ll give up. but, if someone could be so kind as to give me a crash course/walk me through adding one piece of content - in literally the simplest terms possible - i would love that so much. 

thanks ✌🏻

**Comment by fourstix** [comment]
Hi,
A good place to start is with the Installation and Usage guide.  Basically you want to install the agents and then name your videos according to the pattern:  (Studio) - filename (date) and everything works magically.

https://github.com/CodyBerenson/PGMA-Modernized/blob/master/images/PlexGayMetadataAgents-InstallationandUsageGuide.pdf

**Reply by seatownbttm** [reply]
i have looked through that a number of times - i'm getting caught in figuring out how to know what files are supposed to be named for which videos. does that make sense? again, not even sure i'm aware of what i'm actually trying to o which might be the underlying issue :/

**Comment by fourstix** [comment]
If you are new to Plex, I would recommend starting with the tutorials on how to set up Plex and name your movie files.  This is a good explanation of the naming convention Plex uses to locate information for a movie.
 
https://support.plex.tv/articles/naming-and-organizing-your-movie-media-files/ 

There are lots of articles with basic Plex information available at https://support.plex.tv/articles/ . You might want to browse through those, and then come back to the PGMA Installation and Usage guide.

**Reply by seatownbttm** [reply]
cool, thanks for sharin resources. i'll check those out!

**Comment by JPH71** [comment]
First separate you full length films from your scenes.. put them in
different folders.

You name your films and videos using this basic format, taking into account
brackets and spaces as the agent uses these to determine what the studio,
title and release years are so that it can match them against the website
you choose to get the information from.
The basic format is (Studio Name) - Film or video Title (YYYY)... where
yyyy is a 4 year digit like 2022...

Go to a website like AEBN.net and see how the film is represented

You then download the code from the repository and extract the files to the
plug in directory

Quit plex media server then reload it. Click on the spanner symbol at the
top left to go to settings, then click on agents.

Find GayAdult... under it you will see a lot of agents both film and scene
ones... open the settings for each agent then click save to register the
preferences.

Now assuming that you are scraping titles in a film folder, select the Gay
Adult Films agent and select AEBN as you know your film is on that site...

Important... RESTART PLEX

Then go to the home page and select your server...

Create a new library for films, point it to the folder where you are going
to store your films

Then on advance, select GayAdultFilms as your agent. And click OK to create
the library.

If there are film files in the folder plex will retrieve the files then
start downloading the scraped information from the Aebn site.

That's it ....

As you get used to the scraper you may decide you do not want to create
collections based on Genres, countries, directors, actors and scrapers...
you can turn these off... by clicking on the spanner and going to the AEBN
agent settings and turning them off there.. A collection is basically a
grouping of titles, that allows you to see all films for example than an
actor is cast in...

If you have any more questions feel free to ask, we all have to start with
scraping our first film...

If there are any failures, refresh metadata on one film and send the logs..

Scraping will only work if you save the preferences/settings of the agent
you are using...

**Comment by vampirelayer** [comment]
Maybe take a video file you have and we can take it from there!
Am a big fan of these plugins. The team is doing great work.

**Comment by seatownbttm** [comment]
y’all are great!! i think i know where i’m getting caught up - i don’t know how to look at the files i downloaded in the zip file and know what scene or movie they are… how can i tell?

**Comment by vampirelayer** [comment]
Those files in the zip archive are the Plex plugins you need to extract
meta information from different directories so that your video collection
is properly annotated in Plex. The alternative is to do it one by one
manually, which is not ideal if you run into 500 videos like me.

- have you downloaded Plex?
- have you read the installation manual? It can be found here.
[image: image.png]

**Reply by seatownbttm** [reply]
plex is downloaded and i’ve been looking at the manual. 

maybe i’ll try this… i’m just gonna go ahead and ask: what exactly does this plug-in do…? maybe i’m just confused?

**Reply by ms52538** [reply]
It may be quite beneficial for you to take a step back and explain your level of understanding of Plex, content management, meta scraping and meta matching. People would like to assist you but they can't teach you the basics of Plex. You have to learn that for yourself and gain experience thru trial and error before you can understand what this plug-in does. This plug-in does not give you access to content from the internet. This plug-in allows you to meta match your pre-existing content on your computer's Plex installation to specific niche' internet matching forums. It's not content you can download, it's not streaming, it's not torrenting. It's meta matching your pre-existing content on your PC. Please let us know.

**Reply by qboySF** [reply]
So you said that this plug-in does not give you access to content from the internet.  This may not be the right place to ask this, but is there a plug-in that does?  In other words, so that you could add like an Adult or Gay Porn category or something in the main Plex area with a selection of movies similar to the other catgories?  I think I can guess the answer, but sometimes in life there are occasional pleasant surprises.  Very sorry if this is not where I should be asking this.

**Comment by vampirelayer** [comment]
in very laypeople's terms, the plugins make your video collection look like
Netflix, with title, synopsis, genres, cast ...
Plex can do that already with regular films. :)

**Reply by seatownbttm** [reply]
that was EXACTLY what i needed to know. THANK YOU. 💜💜💜

**Comment by JPH71** [comment]
Have you managed to scrape any titles?

Jason x

**Comment by JPH71** [comment]
Any news - did you get stuff to work?
let me know and I can walk you through - step by step....

**Comment by JPH71** [comment]
_PGMA and all the .bundle folders go into the plug in folder... the rest
are unnecessary...

Exit plex media server and restart to activated the new bundles...

Save the preferences for each agent then restart the plex media server
again...

You should be ready to go...

**Comment by JPH71** [comment]
Is all good???

**Comment by JPH71** [comment]
Just checking to see if you managed to resolve your issues???

**Comment by JPH71** [comment]
Let me explain what the plug in does.

You take films that you have downloaded...

You name them using this format: (Studio) - Film Title (4 Digit Year)

Place the renamed video in the folder that plex uses as it's library source.

Now when you set up the library, plex asked or will need you to select a
scraper to process the renamed films.

These scrapers that is to say the plugins will access the Internet pages
they are related to in order to match the film you renamed.

When a match is found the scraper will download the pertinent info for the
film and store it in the plex database and link it to the renamed film.

As it stands there are two main scrapers. One for full films and one for
video clips.
These main scrapers are composed of a number of website specific
sub-scrapers which go to their respective websites to retrieve the data
needed to show the synopsis, genre, cast list etc of the film.

I hope this helps explain what the plugins aka agents aka scrapers do....

Cheers
Jason
